### PR TITLE
Added Twig templates for table and tabs

### PR DESCRIPTION
**Pull request by adamwojs**
| Question | Answer |
| --- | --- |
| Tickets | \- |
| Bug fix? | no |
| New feature? | yes |
| BC breaks? | no |
| Tests pass? | yes |
| Doc needed? | yes |
| License | [GPL-2.0](https://github.com/ezsystems/ezplatform-admin-ui/blob/master/LICENSE) |

Introduced base Twig templates for most commonly used UI components like tabs or tables. 

### Table

image

```twig
    {% embed '@ezdesign/ui/component/table/table.html.twig' %}
        {% block headline %}
            Headline
        {% endblock %}

        {% block actions %}
            <a href="#" class="btn btn-icon">
                <svg class="ibexa-icon ibexa-icon--small ibexa-icon--create">
                    <use xlink:href="{{ ez_icon_path('create') }}"></use>
                </svg>
            </a>
        {% endblock %}

        {% block thead %}
            <tr>
                <th></th>
                <th>Column A</th>
                <th>Column B</th>
                <th>Column C</th>
            </tr>
        {% endblock %}

        {% block tbody %}
            {% for i in 1..10 %}
                <tr>
                    <td></td>
                    <td>A{{ i }}</td>
                    <td>B{{ i }}</td>
                    <td>C{{ i }}</td>
                </tr>
            {% endfor %}
        {% endblock %}
    {% endembed %}
```

#### Supported blocks

<table><tbody><tr><td>&nbsp;</td><td>Block name</td><td>Description</td></tr><tr><td>.</td><td><code>header</code></td><td>&nbsp;</td></tr><tr><td>├──</td><td><code>headline</code></td><td>Table name</td></tr><tr><td>└──</td><td><code>actions</code></td><td>Action buttons e.g create, bulk delete</td></tr><tr><td>.</td><td><code>table</code></td><td>&nbsp;</td></tr><tr><td>├──</td><td><code>thead</code></td><td>Table header content</td></tr><tr><td>└──</td><td><code>tbody</code></td><td>Table body content</td></tr></tbody></table>

#### Supported variables

<table><tbody><tr><td>&nbsp;</td><td>Variable</td><td>Description</td></tr><tr><td>&nbsp;</td><td><code>table_class</code></td><td>Additional CSS classes attached to <code>&lt;table&gt;</code> tag</td></tr></tbody></table>

#### `{% embed ... %}`

image

```twig
 {% embed '@ezdesign/ui/component/tab/tabs.html.twig' with {
        tabs: [
            { id: 'foo', label: 'Foo' },
            { id: 'bar', label: 'Bar' },
        ]
    } %}
        {% block tab_content %}
            {% embed '@ezdesign/ui/component/tab/tab_pane.html.twig' with { id: 'foo', active: true } %}
                {% block content %}
                    Foo tab
                {% endblock %}
            {% endembed %}

            {% embed '@ezdesign/ui/component/tab/tab_pane.html.twig' with { id: 'bar' } %}
                {% block content %}
                    Bar tab. <p>Some <b>Rich</b> HTML <a href="#">content</a></p>
                {% endblock %}
            {% endembed %}
        {% endblock %}
    {% endembed %}
```

#### `{{ include ... }}`

image

```twig
    {% include '@ezdesign/ui/component/tab/tabs.html.twig' with {
        tabs: [
          { id: 'foo', label: 'Foo', content: 'Foo content' },
          { id: 'bar', label: 'Bar', content: 'Bar content', active: true },
        ]
    } %}
```

#### Supported blocks

<table><tbody><tr><td>&nbsp;</td><td>Block name</td><td>Description</td></tr><tr><td>.</td><td><code>tab_content</code></td><td>Tabs content</td></tr></tbody></table>

#### Supported variables

<table><tbody><tr><td>&nbsp;</td><td>Variable</td><td>Description</td></tr><tr><td>.</td><td><code>tabs</code></td><td>&nbsp;</td></tr><tr><td>├──</td><td><code>id</code></td><td>Tab id</td></tr><tr><td>├──</td><td><code>label</code></td><td>Human readable label for tab</td></tr><tr><td>├──</td><td><code>active</code></td><td><code>true</code> if tab is active</td></tr><tr><td>└──</td><td><code>content</code></td><td>HTML content of tab if <code>tab_content</code> is not overriden,&nbsp;</td></tr><tr><td>.</td><td><code>tab_content_class</code></td><td>Additional CSS classes attached to .tab-content</td></tr><tr><td>.</td><td><code>tab_content_attributes</code></td><td>Additional HTML attributes added to .tab-content e.g. <code>{ id: 'example-id', style: '1px solid red' }</code></td></tr></tbody></table>

### When to use {% include %} instead of {% embed %} syntax ?

If tab content is passed as variable while rendering template.

#### Checklist:

*   [x] Coding standards (`$ composer fix-cs`)
*   [x] Ready for Code Review

### CODE REVIEW

**Review by alongosz — APPROVED** [review]

**Review by ViniTou — APPROVED** [review]

**Review by ciastektk — APPROVED** [review]

**Review by webhdx — APPROVED** [review]

**Review by Steveb-p — APPROVED** [review]
Can we make those templates spaceless as well?

While for network transfer it usually doesn't make a difference, it will bloat the output buffer unnecessarily.

We don't usually do that in templates, so it's only a suggestion that we start doing it.

File: src/bundle/Resources/views/themes/admin/ui/tabs.html.twig [hunk 1]
```diff
@@ -0,0 +1,13 @@
+<ul class="nav nav-tabs ez-tabs">
```
Comment (tischsoic) [inline src/bundle/Resources/views/themes/admin/ui/tabs.html.twig:1]:
```suggestion
<ul class="nav nav-tabs ez-tabs" role="tablist">
```

File: src/bundle/Resources/views/themes/admin/ui/tabs.html.twig [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+    {% for tab in tabs %}
+        <li class="nav-item ez-tabs__tab ez-tabs__tab--primary">
+            <a id="ez-tab-label-{{ tab.id }}" href="#ez-tab-{{ tab.id }}" data-toggle="tab" class="nav-link {% if loop.first %}active{% endif %}">
+                {{ tab.label }}
+            </a>
+        </li>
+    {% endfor %}
+</ul>
+
+<div class="tab-content ez-tab-content">
```
Comment (tischsoic) [inline src/bundle/Resources/views/themes/admin/ui/tabs.html.twig:11]: This template should allow adding additional css classes and other attributes
```suggestion
<div class="tab-content ez-tab-content {{ tab_content_class|default('') }}" {{ tab_content_attributes|default('') }}>
```

File: src/bundle/Resources/views/themes/admin/ui/tabs.html.twig [hunk 3]
```diff
@@ -0,0 +1,13 @@
+<ul class="nav nav-tabs ez-tabs">
+    {% for tab in tabs %}
+        <li class="nav-item ez-tabs__tab ez-tabs__tab--primary">
+            <a id="ez-tab-label-{{ tab.id }}" href="#ez-tab-{{ tab.id }}" data-toggle="tab" class="nav-link {% if loop.first %}active{% endif %}">
```
Comment (tischsoic) [inline src/bundle/Resources/views/themes/admin/ui/tabs.html.twig:4]: It seems to be inconsistent to set first tab as active here and active tab-pane based on passed parameter in `/themes/admin/ui/tab.html.twig`.

File: src/bundle/Resources/views/themes/admin/ui/tab.html.twig [hunk 4]
```diff
@@ -0,0 +1,3 @@
+<div id="ez-tab-{{ id }}" class="tab-pane ez-tab-content__pane {% if active|default(false) %}active{% endif %}">
```
Comment (tischsoic) [inline src/bundle/Resources/views/themes/admin/ui/tab.html.twig:1]: For me, the name of this file is misleading, a tab is a `<li class="nav-item ez-tabs__tab...` and not a `tab-pane` so I would change filename to `tab_pane.html.twig`

Comment (tischsoic) [reply src/bundle/Resources/views/themes/admin/ui/tab.html.twig:1]:
```suggestion
<div id="ez-tab-{{ id }}" class="tab-pane ez-tab-content__pane {% if active|default(false) %}active{% endif %} "role="tabpanel" aria-labelledby="ez-tab-label-{{ id }}">
```

File: src/bundle/Resources/views/themes/admin/ui/tabs.html.twig (hunk 3 above)
Comment (tischsoic) [reply src/bundle/Resources/views/themes/admin/ui/tabs.html.twig:4]: Using `tab.id` in `<a>` `id` seems to be right, but for some reason (wrong reason I guess) we are using `group` in `<a>` `id` in many places:
- https://github.com/ezsystems/ezplatform-admin-ui/blob/ed9fdfe629d107c6e464382beab536daac11df13/src/bundle/Resources/views/themes/admin/ui/tab/location_view.html.twig#L8
- https://github.com/ezsystems/ezplatform-admin-ui/blob/ed9fdfe629d107c6e464382beab536daac11df13/src/bundle/Resources/views/themes/admin/ui/tab/default.html.twig#L6

Comment (tischsoic) [reply src/bundle/Resources/views/themes/admin/ui/tabs.html.twig:4]: And now, I realised that we have: https://github.com/ezsystems/ezplatform-admin-ui/blob/ed9fdfe629d107c6e464382beab536daac11df13/src/bundle/Resources/views/themes/admin/ui/tab/default.html.twig so something which we try to create here? 
ping @dew326 @webhdx

File: src/bundle/Resources/views/themes/admin/ui/table.html.twig [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+        </div>
+
+        <div class="ez-table-header__tools">
+            {% block actions %}{% endblock %}
+        </div>
+    </div>
+{% endblock %}
+
+{% block table %}
+    <table class="table table-hover ez-table">
```
Comment (tischsoic) [inline src/bundle/Resources/views/themes/admin/ui/table.html.twig:14]: We should probably allow adding additional CSS classes.
```suggestion
    <table class="table table-hover ez-table {{ table_class|default('') }}">
```

**Review by tischsoic — CHANGES_REQUESTED** [review]

**Comment by dew326** [comment]
We have `/ui/component/` I think it is a more suitable place for those templates.
In addition, I would split them into `/table/` `/tab/`. To avoid ending up with many twigs in one place.

File: src/bundle/Resources/views/themes/admin/ui/tabs.html.twig (hunk 3 above)
Comment (adamwojs) [reply src/bundle/Resources/views/themes/admin/ui/tabs.html.twig]: Propose of [/src/bundle/Resources/views/themes/admin/ui/tab/default.html.twig](https://github.com/ezsystems/ezplatform-admin-ui/blob/ed9fdfe629d107c6e464382beab536daac11df13/src/bundle/Resources/views/themes/admin/ui/tab/default.html.twig) is diffrent: it's default template rendered by  `\EzSystems\EzPlatformAdminUi\Component\TabsComponent` rather then something designed to be reusable :wink:

**Comment by adamwojs** [comment]
PR updated according to code review suggestions.

**Pull request merged**